Modules/cmathmodule.c

```
@@ -36,6 +36,15 @@ class Py_complex_protected_return_converter(CReturnConverter):
         data.return_conversion.append("""
 if (errno == EDOM) {
     PyErr_SetString(PyExc_ValueError, "math domain error");
+
+    PyObject *exc = PyErr_GetRaisedException();
+    PyObject *value = PyComplex_FromCComplex(_return_value);
+
+    if (!value || PyObject_SetAttrString(exc, "value", value)) {
+        PyErr_WriteUnraisable(NULL);
```

Maybe `FormatUnraisable` + some message saying we ignored the exception here?

---

This is absolutely wrong use of `PyErr_WriteUnraisable()`/`PyErr_FormatUnraisable()`. They should only be used if the error cannot be signalled back to the caller (for example if it happened in the garbage collector or in `Py_DECREF()`).

---

Oh. Thank you, TIL. It's not for exceptions that I wanted to be suppressed but still reported. (I only used it once in my life!)

---

Ok, I changed exception handling.